### PR TITLE
Added function opening JSON file with tasks

### PR DESCRIPTION
**Pull request by bartczak-pa**
## Summary by Sourcery

Add a new function to manage opening and reading a JSON file with tasks, along with comprehensive unit tests to ensure robust error handling and correct functionality.

New Features:
- Introduce a function to open a JSON file containing tasks, returning the tasks as a dictionary.

Tests:
- Add unit tests for the new function to handle various scenarios, including valid JSON content, empty JSON, non-existent file, JSON decoding errors, permission errors, and IO errors.

### CODE REVIEW

File: tests/test_file_management.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    [
+        (False, None, {}),
+        (True, "", {}),
+        (True, '{"task1": "do something"}', {"task1": "do something"}),
+        (
+            True,
+            '{"task1": "do something", "task2": "do something else"}',
+            {"task1": "do something", "task2": "do something else"},
+        ),
+    ],
```
[the remark this answers is not in the document]
Comment (bartczak-pa) [reply tests/test_file_management.py:20]: What exceptions I may encounter?

**Comment by bartczak-pa** [comment]
@sourcery-ai review

File: task_cli/file_management.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+from pathlib import Path
+
+
+def open_tasks_file() -> dict:
+    """Open the tasks file and return the tasks dictionary."""
+    tasks: dict = {}
+    tasks_file = Path("tasks.json")
+
+    try:
+        if not tasks_file.exists():
```
[the remark this answers is not in the document]
Comment (bartczak-pa) [reply task_cli/file_management.py:13]: can you explain this in more details?
